static/app/views/dashboardsV2/controls.tsx

```diff
@@ -118,17 +121,25 @@ class Controls extends React.Component<Props> {
                 {t('Edit Dashboard')}
               </Button>
               {organization.features.includes('widget-library') ? (
-                <Button
-                  data-test-id="add-widget-library"
-                  priority="primary"
-                  icon={<IconAdd isCircled />}
-                  onClick={e => {
-                    e.preventDefault();
-                    onAddWidget();
-                  }}
+                <Tooltip
+                  title={tct('Max widgets ([maxWidgets]) per dashboard reached.', {
+                    maxWidgets: MAX_WIDGETS,
+                  })}
+                  disabled={!!!(widgetCount >= MAX_WIDGETS)}
```

Comment:
Similar to my other comment about these fields on the modal, but maybe this is more concise so I'm cool with keeping it. Could we turn the `!!!` into `!` because the comparison is going to always be a boolean? Or we could even flip the operator to be `widgetCount < MAX_WIDGETS`

Reply:
Since `!!!` and `!` are equivalent, I'm just going to leave this as is just because the `disabled` condition for the tooltip is the NOT of the `disabled` condition of the button component.